packages/components/selectFileUploader/SelectFileUploader.web.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { FC, SyntheticEvent, useRef, useState } from "react";
+import { TouchableOpacity, View } from "react-native";
+
+import { SelectFileUploaderProps } from "./SelectFileUploader.type";
+import { formatFile } from "./formatFile";
+import addSVG from "../../../assets/icons/add-circle.svg";
+import { useFeedbacks } from "../../context/FeedbacksProvider";
+import {
+  gradientColorBlue,
+  gradientColorDarkerBlue,
+  gradientColorTurquoise,
+  neutral17,
+  primaryColor,
+  withAlpha,
+} from "../../utils/style/colors";
+import { fontSemibold14 } from "../../utils/style/fonts";
+import { layout } from "../../utils/style/layout";
+import { BrandText } from "../BrandText";
+import { DeleteButton } from "../FilePreview/DeleteButton";
+import { SVGorImageIcon } from "../SVG/SVGorImageIcon";
+import { Box } from "../boxes/Box";
+import { Label } from "../inputs/TextInputCustom";
+
+export const SelectFileUploader: FC<SelectFileUploaderProps> = ({
+  label,
+  style,
+  onUpload,
+  // multiple is not used at true for now, needs to refactor in parents
+  multiple,
+  mimeTypes,
+  children,
+  maxUpload,
+  isImageCover,
+  fileHeight = 256,
+  setIsLoading,
+}) => {
+  const { setToastError } = useFeedbacks();
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
+  const [file, setFile] = useState("");
+
+  const handleFiles = async (files: File[]) => {
+    const _files = multiple ? files : [files[0]];
+    let supportedFiles = [...files].filter(
+      (file) => mimeTypes?.includes(file.type),
+    );
+
+    if (maxUpload && supportedFiles.length) {
+      supportedFiles = supportedFiles.slice(0, maxUpload);
+    }
+
+    if (supportedFiles.length === 0) {
+      setToastError({
+        title: "Unsupported file type.",
+        message: "Sorry we couldn't upload file.",
+      });
+      return;
+    } else if (multiple && supportedFiles.length !== _files.length) {
+      setToastError({
+        title: "Unsupported file type.",
+        message: "Sorry we couldn't upload some files at the moment.",
+      });
+    }
+    if (!multiple) {
+      setFile(URL.createObjectURL(_files[0]));
+    }
+
+    const formattedFiles = await Promise.all(
+      supportedFiles.map(async (file) => await formatFile(file)),
+    );
+
+    onUpload(formattedFiles);
+  };
+
+  const handleChange = async (event: SyntheticEvent) => {
+    setIsLoading?.(true);
+    const targetEvent = event.target as HTMLInputElement;
+
+    if (targetEvent.files && targetEvent.files[0]) {
+      await handleFiles(targetEvent?.files as unknown as File[]);
+    }
+    setIsLoading?.(false);
+  };
+
+  const handleClick = () => {
+    hiddenFileInput?.current?.click?.();
+  };
+
+  const dropHandler = async (ev: any) => {
+    setIsLoading?.(true);
+    ev.preventDefault();
+    if (ev.dataTransfer.items) {
+      const files = [...ev.dataTransfer.items]
```

Comment:
can we just use ```ev.dataTransfer.items.filter ... ``` ? as we don't modify the item so I think it's safe to use directly the array